src/assets/scripts/client/ui/TrafficRateController.js

```diff
@@ -0,0 +1,325 @@
+                <span class="form-value">${rate}</span>
+            </div>`;
+        const $element = $(template);
+        const onChangeHandler = onChangeMethod.bind(this);
+
+        $element.on('change', { rateKey: data }, onChangeHandler);
+
+        return $element;
+    }
+
+    /**
+     * Build form element
+     *
+     * @for TrafficRateController
+     * @method _buildInputField
+     * @param key {string}
+     * @param data {string|object} passed to the change handler
+     * @param onChangeMethod {function}
+     * @return {jquery|HTML Element}
+     */
+    _buildInputField(key, data, onChangeMethod) {
+        const rate = this._rates[key];
+        const name = key.replace(/\./g, ' ');
+        const template = `
+            <div class="form-element">
+                <div class="form-label">${name}</div>
+                <input class="form-input" type="number" name="${key}" value="${rate}" min="0" max="60">
+                <span class="form-value">${rate}</span>
+            </div>`;
+        const $element = $(template);
+        const onChangeHandler = onChangeMethod.bind(this);
+
+        $element.on('change', { rateKey: data }, onChangeHandler);
+
+        return $element;
+    }
+
+    /**
+     * Called when the rate for a flight category was changed
+     *
+     * @for TrafficRateController
+     * @method _onChangeFlightCategoryRate
+     * @param event
+     */
+    _onChangeFlightCategoryRate(event) {
+        const $target = $(event.target);
+        const $output = $target.next(`.${CLASSNAMES.FORM_VALUE}`);
+        const value = $target.val();
+        const category = event.data.rateKey;
+        const elements = this._elements[category];
+
+        this._rates[category] = parseFloat(value);
+
+        $output.text(value);
+
+        for (const { spawnPattern, $formElement } of elements) {
+            this._updateRate(spawnPattern, $formElement);
+        }
+    }
+
+    /**
+     * Called when the rate for a route was changed
+     *
+     * @for TrafficRateController
+     * @method _onChangeSpawnPatternRate
+     * @param event
+     */
+    _onChangeSpawnPatternRate(event) {
+        const $target = $(event.target);
+        const value = $target.val();
+        const spawnPattern = event.data.rateKey;
+
+        this._rates[spawnPattern.routeString] = parseFloat(value);
+
+        this._updateRate(spawnPattern);
+    }
+
+    /**
+     * Recalculate the rate for a spawn pattern
+     *
+     * @for TrafficRateController
+     * @method _updateRate
+     * @param spawnPattern {SpawnPatternModel}
+     * @param $formElement {jQuery element}
+     */
+    _updateRate(spawnPattern, $formElement) {
+        const { category, routeString } = spawnPattern;
+        const $output = $formElement.children(`.${CLASSNAMES.FORM_VALUE}`);
+
+        spawnPattern.rate = this._rates[category] * this._rates[routeString];
+
+        $output.text(spawnPattern.rate);
+
+        SpawnScheduler.resetTimer(spawnPattern);
```

Comment:
When the rate is updated, perhaps we can do something of a replacement of the existing arrivals along that spawn pattern which haven't reached the airspace yet?

Basically, we could start with just removing all aircraft and re-running the prespawn logic, and it should be a good first step.

Reply:
Right now, I am calculating the time since the last aircraft was spawned and then either spawn a new aircraft or create a new timer with the difference (= newTimeBetweenAircrafts - actualTimePassedSinceLastSpawn)

Reply:
I like @felixscheffer approach here.  I feel it would be less arresting to the user.  Modifying existing aircraft would be like when you're playing a multi-player game and the server response is slow.  I don't think it would be a good experience.

Reply:
When the spawn points are super far away from the airspace boundaries though, it could (theoretically) take _hours_ for the new spawn rate to be noticed by the controller. That's why I'd be inclined to drop anybody outside of the airspace and regenerate them by pre-spawning. That way, the new rates would very quickly take effect, and makes it more useful as a training tool.

Reply:
Created #1343 for this as a future request-- this is good for now. Resolving.